application/src/main/java/org/togetherjava/tjbot/commands/tags/TagCommand.java

```diff
@@ -46,20 +50,35 @@ public void onSlashCommand(SlashCommandInteractionEvent event) {
         String id = Objects.requireNonNull(event.getOption(ID_OPTION)).getAsString();
         OptionMapping replyToUserOption = event.getOption(REPLY_TO_USER_OPTION);
 
-        if (tagSystem.handleIsUnknownTag(id, event)) {
+        if (tagSystem.handleIsUnknownTag(id, event, super.getComponentIdGenerator(),
+                replyToUserOption)) {
             return;
         }
 
-        ReplyCallbackAction message = event
-            .replyEmbeds(new EmbedBuilder().setDescription(tagSystem.getTag(id).orElseThrow())
-                .setFooter(event.getUser().getName() + " • used " + event.getCommandString())
+        sendTagReply(event, event.getUser().getName(), id, event.getCommandString(),
+                replyToUserOption == null ? null : replyToUserOption.getAsUser().getAsMention());
+    }
+
+    @Override
+    public void onButtonClick(ButtonInteractionEvent event, List<String> args) {
+        sendTagReply(event, event.getUser().getName(), args.get(0), null, args.get(1));
+    }
+
+    /**
+     * Sends the reply for a successfull /tag use (i.e. the given tag exists)
+     */
+    private void sendTagReply(IReplyCallback callback, String userName, String tag,
+            @Nullable String commandString, @Nullable String replyToUser) {
+        Optional<String> commandStringOpt = Optional.ofNullable(commandString);
+        Optional<String> replyToUserOpt = Optional.ofNullable(replyToUser);
+
+        callback
+            .replyEmbeds(new EmbedBuilder().setDescription(tagSystem.getTag(tag).orElseThrow())
+                .setFooter(userName + commandStringOpt.map(s -> " • used " + s).orElse(""))
                 .setTimestamp(Instant.now())
                 .setColor(TagSystem.AMBIENT_COLOR)
-                .build());
-
-        if (replyToUserOption != null) {
-            message = message.setContent(replyToUserOption.getAsUser().getAsMention());
-        }
-        message.queue();
+                .build())
+            .setContent(replyToUserOpt.orElse(""))
```

Comment:
dont know the javadoc of `setContent`, but many JDA methods accept `null`. If thats the case, this could be a nice improvement, since u dont need the optional anymore then